lua/trouble/providers/diagnostic.lua

```diff
@@ -31,7 +31,7 @@ function M.diagnostics(_, buf, cb, options)
   end
 
   local messages = {}
-  if options.severity ~= nil then
+  if options.severity[options.severity] then
```

Comment:
what does this even do? Did you test this?

It uses the table itself as a key in the table?